src/components/Project/ProjectHeader/ProjectFavouriteField.tsx

```diff
@@ -1,18 +1,22 @@
-import { FC, MouseEvent } from 'react';
+import { FC, MouseEvent, useCallback } from 'react';
 import { Control, Controller, FieldValues } from 'react-hook-form';
+import { useAppSelector } from '@/hooks/common';
 import { IconStar, IconStarFill } from 'hds-react/icons';
 import { IconButton } from '../../shared';
 import { IProjectHeaderFieldProps } from './ProjectHeader';
+import { selectMode } from '@/reducers/projectSlice';
 
 const ProjectFavouriteField: FC<IProjectHeaderFieldProps> = ({ control }) => {
-  const handleClick = (
-    e: MouseEvent<HTMLButtonElement>,
-    onChange: (...event: unknown[]) => void,
-    value: string,
-  ) => {
-    e.preventDefault();
-    onChange(!value);
-  };
+  const projectMode = useAppSelector(selectMode);
+  const handleClick = useCallback(
+    (e: MouseEvent<HTMLButtonElement>, onChange: (...event: unknown[]) => void, value: string) => {
+      e.preventDefault();
+      if (projectMode === 'edit') {
```

Comment:
instead of changing the click could we add the disabled prop to IconButton?

interface IIconButton {
   ...
  disabled?: boolean;
}

const IconButton: FC<IIconButton> = (props) => {
  const { onClick, disabled, ...iconProps } = props;
  return (
    <button
      onClick={onClick}
      aria-label={iconProps.text || iconProps.label}
      disabled={disabled || false}
    >
    ...